packages/core/rollup.config.mjs

```diff
@@ -0,0 +1,62 @@
+// The following is only possible from Node 18 onwards
+// import pkg from "./package.json" assert { type: "json" };
+
+// Until we only support Node 18+, this should be used instead
+// (see https://rollupjs.org/guide/en/#importing-packagejson) 
+import { createRequire } from 'node:module';
+const require = createRequire(import.meta.url);
+const pkg = require('./package.json');
+
+import typescript from "rollup-plugin-typescript2";
+
+export default [{
+  input: "./src/index.ts",
+  output: [
+    {
+      file: pkg.main,
+      format: "cjs",
+    },
+    {
+      file: pkg.module,
+      format: "esm",
+    },
+  ],
+  plugins: [
+    typescript({
+      // Use our own version of TypeScript, rather than the one bundled with the plugin:
+      typescript: require("typescript"),
+      tsconfigOverride: {
+        compilerOptions: {
+          module: "esnext",
+        },
+      },
+    })
+  ],
+  // The following option is useful because symlinks are used in monorepos
+  preserveSymlinks: true,
```

Comment:
Let's move this into a `common` const, as it's used in both exports (index & mocks)